test_plans/group.asciidoc

```diff
@@ -0,0 +1,71 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for group class
+
+This is a test plan for the group class as described in Section 4.9.1.7. of the SYCL 2020 specification.
+
+Estimated development time is two and a half days.
+
+== Testing scope
+
+=== Backend coverage
+
+All the tests described below are not backend-specific and are performed for any SYCL backend.
+
+=== Device coverage
+
+All tests construct a test device for which conformance is assessed. All the tests described below are performed once for that test device.
+
+== Tests
+
+=== Existing test
+The existing test in `group/group_api.cpp` defines some greater than zero group counts (2, 4, 8) with greater than zero group sizes (4, 3, 2), in three dimensions. A kernel is dispatched that calls the below functions, gathers the results for one work-item in each work-group and checks the return types.
+
+- `group.get_id().get(int)`
+- `group.get_id(int)`
+- `group.get_local_range().get(int)`
+- `group.get_local_range(int)`
+- `group.get_global_range().get(int)`
+- `group.get_global_range(int)`
+- `group.get_group_range().get(int)`
+- `group.get_group_range(int)`
+- `group.operator[]`
+
+After the terminates, the results and the return type check is verified.
+
+=== Modification
+
+The test is modified in the following way. The results are gathered for each work-item in every work-group, instead of for one work-item in every work-group. Furthermore, the expected results are initialized to some special value (not zero), to make sure the value is overwritten by the kernel. The following function tests are marked as deprecated:
+
+- `group.get_id().get(int)`
+- `group.get_id(int)`
+- `group.get_global_range().get(int)`
+- `group.get_global_range(int)`
```

Comment:
None of these functions exist in the SYCL 2020 specification, not even as deprecated.  I agree that they were all defined in SYCL 1.2.1.

I was not part of the WG when this decision was made, so I don't know if it was an oversight.  Do other members of the WG think that we intentionally removed these functions (vs. deprecating them)?  Or, was this an oversight?

If we intentionally removed them from the spec, then the CTS should not test them at all.

Reply:
I think we should write the CTS to the current state of the SYCL 2020 spec.  Since the functions I list above are not in the SYCL 2020 spec, we should remove testing for them from the CTS.

Reply:
Okay that makes sense, I have changed it in the test plan.